In [4]:
import pandas as pd
import numpy as np
from scipy import stats
df=pd.read_csv(r'/content/sales_data_with_discounts.csv')
df

Date       Day  SKU City  Volume         BU  Brand       Model  \
0    01-04-2021  Thursday  M01    C      15    Mobiles  RealU       RU-10   
1    01-04-2021  Thursday  M02    C      10    Mobiles  RealU   RU-9 Plus   
2    01-04-2021  Thursday  M03    C       7    Mobiles   YouM       YM-99   
3    01-04-2021  Thursday  M04    C       6    Mobiles   YouM  YM-99 Plus   
4    01-04-2021  Thursday  M05    C       3    Mobiles   YouM      YM-98    
..          ...       ...  ...  ...     ...        ...    ...         ...   
445  15-04-2021  Thursday  L06    C       2  Lifestyle  Jeera   M-Casuals   
446  15-04-2021  Thursday  L07    C       6  Lifestyle   Viva   W-Western   
447  15-04-2021  Thursday  L08    C       2  Lifestyle   Viva    W-Lounge   
448  15-04-2021  Thursday  L09    C       3  Lifestyle  Jeera   M-Formals   
449  15-04-2021  Thursday  L10    C       1  Lifestyle  Jeera     M-Shoes   

     Avg Price  Total Sales Value  Discount Rate (%)  Discount Amount  \
0        12100             181500          11.654820     21153.498820   
1        10100             101000          11.560498     11676.102961   
2        16100             112700           9.456886     10657.910157   
3        20100             120600           6.935385      8364.074702   
4         8100              24300          17.995663      4372.946230   
..         ...                ...                ...              ...   
445       1300               2600          15.475687       402.367873   
446       2600              15600          17.057027      2660.896242   
447       1600               3200          18.965550       606.897606   
448       1900               5700          16.793014       957.201826   
449       3100               3100          15.333300       475.332295   

     Net Sales Value  
0      160346.501180  
1       89323.897039  
2      102042.089843  
3      112235.925298  
4       19927.053770  
..               ...  
445      2197.632127  
446     12939.103758  
447      2593.102394  
448      4742.798174  
449      2624.667705  

[450 rows x 13 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 450 entries, 0 to 449
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Date               450 non-null    object 
 1   Day                450 non-null    object 
 2   SKU                450 non-null    object 
 3   City               450 non-null    object 
 4   Volume             450 non-null    int64  
 5   BU                 450 non-null    object 
 6   Brand              450 non-null    object 
 7   Model              450 non-null    object 
 8   Avg Price          450 non-null    int64  
 9   Total Sales Value  450 non-null    int64  
 10  Discount Rate (%)  450 non-null    float64
 11  Discount Amount    450 non-null    float64
 12  Net Sales Value    450 non-null    float64
dtypes: float64(3), int64(3), object(7)
memory usage: 45.8+ KB


In [ ]:
df.shape,df.describe()

((450, 13),
            Volume     Avg Price  Total Sales Value  Discount Rate (%)  \
 count  450.000000    450.000000         450.000000         450.000000   
 mean     5.066667  10453.433333       33812.835556          15.155242   
 std      4.231602  18079.904840       50535.074173           4.220602   
 min      1.000000    290.000000         400.000000           5.007822   
 25%      3.000000    465.000000        2700.000000          13.965063   
 50%      4.000000   1450.000000        5700.000000          16.577766   
 75%      6.000000  10100.000000       53200.000000          18.114718   
 max     31.000000  60100.000000      196400.000000          19.992407   
 
        Discount Amount  Net Sales Value  
 count       450.000000       450.000000  
 mean       3346.499424     30466.336131  
 std        4509.902963     46358.656624  
 min          69.177942       326.974801  
 25%         460.459304      2202.208645  
 50%         988.933733      4677.788059  
 75%        5316.49

In [6]:
#Identify Nemeric columns in dataframe
numeric_columns = df.select_dtypes(include=np.number).columns
#display numeric columns
numeric_columns


Index(['Volume', 'Avg Price', 'Total Sales Value', 'Discount Rate (%)',
       'Discount Amount', 'Net Sales Value'],
      dtype='object')

In [7]:
print('Mean values of  Volume',np.mean(df['Volume']),'  Avg Price',np.mean(df['Avg Price']),'Total Sales Value',np.mean(df['Total Sales Value']))

Mean values of  Volume 5.066666666666666   Avg Price 10453.433333333332 Total Sales Value 33812.83555555555


In [14]:
print('Mean values of  Discount Rate (%)',np.mean(df['Discount Rate (%)']),' Discount Amount',np.mean(df['Discount Amount']),'Net Sales Value',np.mean(df['Net Sales Value']))

Mean values of  Discount Rate (%) 15.155241895330914  Discount Amount 3346.499424342415 Net Sales Value 30466.336131270138


**Interpretation**:
Mean of Net sales value  is 30466.336131270138 and  mean of Total Sales Value is 33812.83555555555, Mean of discount amount is  3346.499424342415.Therefore we interpret that Net Sales value,Total Sales Value are relatively more than discount amount which is good.

In [13]:
print('Median values of  Volume',np.median(df['Volume']),'  Avg Price',np.median(df['Avg Price']),'Total Sales Value',np.median(df['Total Sales Value']))

Median values of  Volume 4.0   Avg Price 1450.0 Total Sales Value 5700.0


In [12]:
print('Median values of  Discount Rate (%)',np.median(df['Discount Rate (%)']),'  Discount Amount',np.median(df['Discount Amount']),'Net Sales Value',np.median(df['Net Sales Value']))

Median values of  Discount Rate (%) 16.57776564929597   Discount Amount 988.9337332382784 Net Sales Value 4677.788059301853


In [11]:
print('Mode values of  Volume',stats.mode(df['Volume']),'  Avg Price',stats.mode(df['Avg Price']),'Total Sales Value',stats.mode(df['Total Sales Value']))

Mode values of  Volume ModeResult(mode=3, count=110)   Avg Price ModeResult(mode=400, count=30) Total Sales Value ModeResult(mode=24300, count=14)


In [10]:
print('Mode values of  Discount Rate (%)',stats.mode(df['Discount Rate (%)']),'  Discount Amount',stats.mode(df['Discount Amount']),'Net Sales Value',stats.mode(df['Net Sales Value']))

Mode values of  Discount Rate (%) ModeResult(mode=5.007822189204133, count=1)   Discount Amount ModeResult(mode=69.17794228822787, count=1) Net Sales Value ModeResult(mode=326.97480055470817, count=1)


Interpretation:
Mode of Discount rate (%) is 5.007%,The mode of 5% discount indicates that the range from 0% to 20% shows flexibility in discounting strategies

In [9]:
print('Standard deviation values of  Volume',np.std(df['Volume']),'  Avg Price',np.std(df['Avg Price']),'Total Sales Value',np.std(df['Total Sales Value']))

Standard deviation values of  Volume 4.226897995772629   Avg Price 18059.804883928166 Total Sales Value 50478.892860940316


In [8]:
print('Standard deviation values of  Discount Rate (%)',np.std(df['Discount Rate (%)']),'  Discount Amount',np.std(df['Discount Amount']),'Net Sales Value',np.std(df['Net Sales Value']))

Standard deviation values of  Discount Rate (%) 4.215909987168583   Discount Amount 4504.889173073149 Net Sales Value 46307.118357300766


**Interpretation**: A standard deviation of Avg Price 18059.804883928166 Total Sales Value 50478.892860940316, Net Sales Value 46307.118357300766 indicates there is variability in Total Sales Value, Net Sales Value potentially influenced by customer preferences or seasonal trends.

By applying descriptive analytics to numerical columns in a sales and discount dataset, businesses can gain actionable insights to optimize pricing, improve promotional strategies, and enhance overall performance. These insights form the basis for informed decision-making and strategic planning in sales management.
In summary, interpreting mean, median, mode, and standard deviation in sales data provides a comprehensive view of sales performance, aiding businesses in understanding central tendencies, variability, and distribution patterns crucial for strategic planning and decision-making.